src/data/Storage.ts

```diff
@@ -0,0 +1,172 @@
+import { Storage, Drivers } from "@ionic/storage";
+import type { Cycle } from "./ClassCycle";
+
+export interface Context {
+    cycles: Cycle[];
+}
+
+const storageImpl = new Storage({
+    name: "PeriodDB",
+    driverOrder: [Drivers.IndexedDB, Drivers.LocalStorage],
+});
+
+storageImpl
+  .create()
+  .then(() => console.log("Storage created"))
+  .catch((err) =>
+      console.error(`Can't create storage ${(err as Error).message}`)
+  );
+
+function setCycles(cycles: Cycle[]): Promise<void> {
+    return storageImpl.set("cycles", cycles);
+}
+
+async function getCycles(safe: boolean): Promise<Cycle[]> {
+    const value = (await storageImpl.get("cycles")) as Cycle[];
+    if (safe && !value) {
+        throw new Error("Can't find `cycles` in storage");
+    }
+    return value;
+}
+
+export const storage = {
+    set: {
+        cycles: setCycles,
+    },
+    get: {
+        cycles: () => getCycles(true),
+    },
+};
+
+// NOTE: Predefined templates for test purpose
+//       for use just uncomment one of the following lines:
+// emptyArrayOfCycles().catch((err) => console.error(err));
+// arrayOfCyclesWithSingleCycle().catch((err) => console.error(err));
+// arrayOfCyclesWithSingleCycleWithDelay().catch((err) => console.error(err));
+// halfFilledArrayOfCycles().catch((err) => console.error(err));
+// fullyFilledArrayOfCycles().catch((err) => console.error(err));
+// fullyFilledArrayOfCyclesWithDelay().catch((err) => console.error(err));
+
+function emptyArrayOfCycles(): Promise<void> {
+    return storageImpl.remove("cycles");
+}
+
+function arrayOfCyclesWithSingleCycle(): Promise<void> {
+    return storage.set.cycles([
+        {
+        cycleLength: 28,
+        periodLength: 6,
+        startDate: "2023-05-31",
+        },
+    ] satisfies Cycle[]);
```

Comment:
I have a mistake: 'Cycle' only refers to a type, but is being used as a value her.
I fixed it when I "satisfies" it to "as"

Reply:
I think you can update your `ts` package, because `satisfies` keyword was introduced in 4.9 version